src/web/app/pages-instructor/instructor-course-enroll-page/instructor-course-enroll-page.component.ts

```diff
@@ -94,59 +105,159 @@ export class InstructorCourseEnrollPageComponent implements OnInit {
+              return;
+            }
+
+            currentStudentChunk.push({
+              section: _section,
+              team: _team,
+              name: _name,
+              email: _email,
+              comments: _comments,
+            });
+            totalStudentCount += 1;
+          }
+
+          if (currentStudentChunk.length >= this.numberOfStudentsPerRequest) {
+            this.allStudentChunks.push(currentStudentChunk);
+            currentStudentChunk = [];
+          }
+        });
+    if (currentStudentChunk.length > 0) {
+      this.allStudentChunks.push(currentStudentChunk);
+      currentStudentChunk = [];
+    }
+    if (this.allStudentChunks.length === 0 || mainErrorFlag) {
+      this.enrollErrorMessage += this.generalErrorMessage;
+      this.statusMessageService.showErrorToast('There are some invalid rows');
+      this.isEnrolling = false;
+      return;
+    }
+
+    const enrolledStudents: Student[] = [];
+
+    // Use concat because we cannot afford to parallelize with forkJoin when there's data dependency
+    const enrollRequests: Observable<Students> = concat(
+        ...this.allStudentChunks.map((studentChunk: StudentEnrollRequest[]) => {
+          const studentsEnrollRequest: StudentsEnrollRequest = {
+            studentEnrollRequests: studentChunk,
+          };
+          return this.studentService.enrollStudents(
+              this.courseId, studentsEnrollRequest,
+          );
+        }),
+    );
+
+    enrollRequests.pipe(finalize(() => this.isEnrolling = false)).subscribe({
+      next: (resp: Students) => {
+        enrolledStudents.push(...resp.students);
+      },
+      complete: () => {
+        this.showEnrollResults = true;
+        this.statusMessage.pop(); // removes any existing error status message
+        this.statusMessageService.showSuccessToast('Enrollment successful. Summary given below.');
+
+        const allStudentEntries: StudentEnrollRequest[] = [];
+        this.allStudentChunks.forEach((chunk: StudentEnrollRequest[]) => allStudentEntries.push(...chunk));
+
+        this.enrollResultPanelList =
+            this.populateEnrollResultPanelList(this.existingStudents, enrolledStudents, allStudentEntries);
+
+        this.studentService.getStudentsFromCourse({ courseId: this.courseId }).subscribe((resp: Students) => {
+          this.existingStudents = resp.students;
+          if (!this.isExistingStudentsPanelCollapsed) {
+            const existingStudentTable: Handsontable = this.hotRegisterer.getInstance(this.existingStudentsHOT);
+            this.loadExistingStudentsData(existingStudentTable, this.existingStudents);
+          }
+          this.isExistingStudentsPresent = true;
+        });
+      },
+      error: (resp: ErrorMessageOutput) => {
+        this.enrollErrorMessage = resp.error.message;
+      },
     });
   }
 
+  private checkCellsValidity(section: string, team: string, name: string, email: string, totalStudentCount: number,
```

Comment:
do we really need to pass in these parameters individually like this? esp when we have an interface defined for those. Plus `totalStudentCount` is unnecessary - there are ways to do the 100 student check without an explicit variable.